### PR TITLE
Infinity and early conditions

### PR DESCRIPTION
**Pull request by customcommander**
This pull request has a few things going on:

1. Mocha was a bit out of date. Upgraded it but that meant having to stop testing on Node.js 6 and below. These versions of Node.js aren't supported anymore so probably not a big deal anyway.
2. Added a few missing tests: numbers aren't rounded _and_ negative numbers produce empty strings
3. There were `var` _and_ `const`. No need to have both so replaced with `const`.
4. Early exit condition when the string to repeat is empty
5. Breaking change: can't repeat a string when number is `Infinity`

**Infinity**

The native `String#repeat` method does throw an error when `Infinity` is given as a parameter. So this changes aligns with the native behaviour. Also there's an issue with how `Infinity` works right now: with the same string to repeat, it will produce a _different_ string:

```
const repeat = require('repeat-string');

repeat('🌯', Infinity)
//=> '🌯🌯'
repeat('🌯', Infinity)
//=> '🌯🌯🌯🌯'
repeat('🌯', Infinity)
//=> '🌯🌯🌯🌯🌯🌯' 
```

This didn't seem right to me so took the liberty to throw an error instead.

### CODE REVIEW

**Comment by customcommander** [comment]
@jonschlinkert 👋

File: index.js [hunk 1]
```diff
@@ -43,11 +43,19 @@ function repeat(str, num) {
...
+  // no need to repeat an empty string
+  if (str === '') return '';
+
   // cover common, quick use cases
   if (num === 1) return str;
   if (num === 2) return str + str;
 
   var max = str.length * num;
+
+  if (max === Infinity) {
```
Comment (jonschlinkert) [inline index.js:55]: Nice work. Please move this check to be just after the string check (line 46 or so) and change it to `if (num === Infinity)`.

**Comment by jonschlinkert** [comment]
@customcommander nice work! I only had the one comment, everything else is great. thank you!

File: index.js (hunk 1 above)
Comment (customcommander) [reply index.js:55]: The reason I put it on that line was to leverage the implicit "parseFloat()" when `str.length` and `num` are multiplied together e.g.

```javascript
5 * Infinity;
//=> Infinity

5 * 'Infinity';
//=> Infinity

5 * '    Infinity   ';
//=> Infinity
```

That was also motivated by some of the tests that seem to suggest that `num` can be given as a string too. I'm happy to make the change but should I still care about the case of `Infinity` as a string then?

By the way it seems that if `num` can be given as a string then the two shortcuts at L50,51 are most likely ignored. Should this be addressed?
